dice.py

```diff
@@ -0,0 +1,35 @@
+import random
+
+
+class Dice:
+    """
+    A dice. That you can roll. Can have arbitrary objects sides.
+    """
+    def __init__(self, sides):
+        """
+        Instantiates a Dice object.
+
+        Arguments:
+        sides: A list of the sides of the dice. Each object in this can be
+        anything (an str, a list of str, etc)
+        """
+
+        if type(sides) is not list:
```

Comment:
Comparing `type()` is super seldom helpful, because it's very literal about what it's looking for and, for example, doesn't accommodate subclassing. To illustrate:
```
from collections import defaultdict
type defaultdict() == dict  # False
```
means that any code checking `type()` for `dict` won't accept the `defaultdict` instance, despite it being a specialised subclass of `dict` that is an appropriate substitute for `dict` in all but the rarest of cases.

`isinstance()` gets over the subclass problem (`isinstance(dedaultdict(), dict)` is `True`) but it still doesn't get to the heart of why you only want lists here. Wherever possible, check either for a specific attribute or, even better, try to do what you want the list to be able to do and catch the exception if you can't. Obviously, in this particular case you probably want to know this on instantiation not when the die is finally rolled, so it's not really asking forgiveness like we like to, but it's still a better test of function than going by class.